### ISSUE TITLE
No import completions from npm packages

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#24267]
No named import completions for module

```
npm init --yes
npm install danger
tsc --init
```

Now create a `.ts` file

```ts
import { danger, fail, } from "danger";

pe/**/
```

Request completions at `/**/`

**Expected**: Completion for `peril`.
**Actual**: No completion.

**Comment by mhegazy** [comment]
There are few issues here.. First module augmentations seems to diable completions alltogather.. here is a simpler repro:

```ts
// c.ts
export var y = 0;
```

```ts
// b.ts
export { };

declare module "./c" {
    export var x;
}
```

```ts
// a.ts

import * as B from "./b";

|  // expect `x` and `y` to show up here. 
```

The other issue here is in the declaration file for `danger` is wrong, and is not flagged as such.. the module is declared as:
```ts
import * as GitHub from "@octokit/rest";
declare module "danger" {
    ...
}
```

which should be an error really, since the module augmentation is invalid.. not sure why we are not erroring there..

the file should rather be written as a module, or as:

```ts
declare module "danger" {
    import * as GitHub from "@octokit/rest";
     ...
}
```

**Comment by DanielRosenwasser** [comment]
Psst @orta

**Comment by mhegazy** [comment]
@andy-ms do you know why was the invalid augmentation error not shown in this file?

**Comment by andy-ms** [comment]
Made a new issue for that 2.9.0-dev.20180519
